src/Axe/Controllers/CompilerController.cs

```diff
@@ -78,17 +91,22 @@ public IActionResult Solve(CodeBlockVm model)
             return View(model);
         }
 
+        [Authorize(Roles = "superuser")]
```

Comment:
[Authorize(Roles = "superuser")]
superuser/admin access is required for some site management task
access to compiler tasks isn't such operation
users attempts to access task editor should be handled (allowed/denied) based on their expert skills:
ICollection<ExpertTechnologyLink> Technologies

Reply:
Do you agree that need to add new role 'expert', then to create custom claims-based authorization tag with technologies for user (expert) and to use it tag for methods of controller? I think, It's flexible solution for this case.

Reply:
expert in c# is not necessary expert in js, for example. if authorization with "expert" role allows such low-level control (allow to create [c#] task, deny [js] tasks access)  - I'm ok with it.
I think there some cases where role="expert" isn't enough, e.g Index operation - return to user all technologies, where user can create a new task. c# expert should be allowed access to Index, but he will see only [c#] and not [js]. authorize attribute can't separate that

Reply:
I can add claim (in which technologies user is expert) to user then use it into custom attribute